### ISSUE TITLE
Documentation is incorrect

### ISSUE THREAD

**Issue by kamransoomro84** [node-red/node-red#4099]
### Current Behavior

The page [here](https://nodered.org/docs/creating-nodes/first-node) states the following:

> To test a node module locally the [npm install <folder>](https://docs.npmjs.com/cli/install) command can be used. This allows you to develop the node in a local directory and have it linked into a local node-red install during development.

Following these instructions the test node did not show up in the palette.

### Expected Behavior

The above information is incorrect as the `settings.json` file states that:

> Node-RED scans the `nodes` directory in the userDir to find local node files.

I have renamed the `node_modules` directory to `node` in `~/.node-red` and can confirm that it works.

**Comment by knolleary** [comment]
> The above information is incorrect as the settings.json file states that:

They are two separate things.

> I have renamed the node_modules directory to node in ~/.node-red and can confirm that it works.

That is not the right thing to have done.

Node-RED scans both the `node_modules` directory *and* the `nodes` directory.

The `node_modules` directory is the one managed by `npm` and is where all of the dependencies listed in your `package.json` are installed. That will include both node-red modules and 'normal' npm modules. By renaming that to `nodes`, the runtime will no longer be able to find any non-node-red modules you want to have installed.

When developing locally, you can *either* `npm install` the module as described in the docs (which will result it being under `node_modules`, *or* you can manually copy/link it under the `nodes` directory. Both will work equally well. We don't document both approaches as it's easier to document one of them and not force the reader to choose an approach.

> Following these instructions the test node did not show up in the palette.

Did you restart Node-RED after installing?

**Comment by kamransoomro84** [comment]
Yes I did restart node-red and even turned on trace logging to see if the node was being detected and there was nothing in the output to indicate that node-red could see the node specification.
